force-app/main/default/classes/DMLOperations.cls

```diff
@@ -97,10 +104,11 @@ public with sharing class DMLOperations {
     public static void updateOpportunityStage(Id oppId, String newStage) {
         // Retrieve the Opportunity from the database
         // [SELECT Id FROM Opportunity WHERE Id = :oppId]
-
+        Opportunity existingOpp = [SELECT Id FROM Opportunity WHERE Id = :oppId];
         // Update the Opportunity's stage
-
+        existingOpp.StageName = newStage;
         // Update the Opportunity in the database
+        update existingOpp;
```

Comment:
Good another technique and you could use this above is is below. You already have the ID. So just update a contact with that ID. 

```
  public static void updateOpportunityStage(Id oppId, String newStage) {
        Opportunity opportunity = new Opportunity(Id = oppId, StageName = newStage);
        update opportunity;
    }
    ```